typedb-src/modules/ROOT/pages/typedb-in-20-queries.adoc

```diff
@@ -0,0 +1,725 @@
+
+.Inherited attributes example
+[,typeql]
+----
+match
+    $u isa user,
+        has name "Bob";
+----
+
+We match a user with name `Bob`.
+The `user` type inherits the ability to own  the `name` attribute type from its supertype -- the `person` type.
+See the schema from the <<_schema,query #1>>.
+
+== #{counter:query}: Override inheritance
+_Run this query after <<_schema,query #1>>._
+
+You can override an inherited ownership or a role with a new name.
+//#todo Add explanation as to why it might be needed
+Override uses the keyword `as` with the new type preceding the keyword, and old (inherited) type following the keyword.
+The new type should be defined in the schema as a subtype of the overriden (inherited) type.
+
+.Schema
+[,typeql]
+----
+define
+
+user owns full-name as name;
+
+user-permission sub permission,
+    relates user as subject;
+----
+
+In the above example, we override the inherited `name` attribute type with the `full-name` attribute type,
+We also subtype `permission` relation type with the `user-permission`, which overrides the inherited `subject` role
+with the `user` role. The `object` role is inherited without overriding in this case.
+
+== #{counter:query}: Modify relations
+_Run this query after <<_schema,query #1>>, <<_data,query #2>>, and <<_extend_schema,query #4>>._
+//n-ary
+//nested
+//variadic (1 role, many role players?)
+
+Relations in TypeDB work elegantly and naturally, without the need for joins,
+tables, foreign keys, or any other tricks needed to "just make it work".
+
+A database schema defines relation types and their roles, as well as types that can play a role.
+Creating a relation is just instantiating a relation type with the exact role players that you want.
+
+We can create an n-ary relation with just one role, for example,
+a `friendship` relation with three role players for the role `friend`.
+
+In <<_extend_schema>> section we extended the schema by adding a third role to the `permission` relation.
+Now let's add a role player for this role for a relation, that was inserted before the extension of the schema:
+
+.Relation example
+[,typeql]
+----
+match
+    $u isa user,
+        has name "Bob";
+    $f isa file,
+        has path "README.md";
+    $p($u, $f) isa permission;
+insert
+    $a isa action, has name "edit";
+    $p(permitted-action:$a);
+----
+
+The above query adds `action` entity with name `edit` to the `permission` for the user with name `Bob`
+and the file with the path `README.md`.
+See the visualization of the result below.
+
+image::typedb-in-20-queries/relation.png[Relation, role=framed, width = 75%, link=self]
+
+== #{counter:query}: Utilize immutable attributes
+_Run this query after <<_schema,query #1>>, <<_data,query #2>>, and <<_data,query #4>>._
+
+Always optimal storage space and memory requirements as data is naturally deduplicated.
+
+Only attributes have values and all attributes are immutable.
+We can't change the value of an attribute,
```

Comment:
I find this sentence misleading. Of course, I can update the value of an attribute?